sw/device/lib/dif/dif_kmac_unittest.cc

```diff
@@ -86,5 +90,90 @@ TEST(FunctionNameTest, BadArg) {
   EXPECT_EQ(dif_kmac_function_name_init("", -1, &fn), kDifBadArg);
 }
 
+using mock_mmio::MmioTest;
+using mock_mmio::MockDevice;
+using testing::Test;
+
+// Base class for the rest fixtures in this file.
+class KmacTest : public testing::Test, public mock_mmio::MmioTest {
+ public:
+  dif_kmac_t kmac_;
+  dif_kmac_operation_state_t op_state_;
+  const uint8_t kMsg_[17] = {0xa7, 0x48, 0x47, 0x93, 0x0a, 0x03,
+                             0xab, 0xee, 0xa4, 0x73, 0xe1, 0xf3,
+                             0xdc, 0x30, 0xb8, 0x88, 0x15};
+  const size_t kLen_ = sizeof(kMsg_);
+
+  KmacTest() { EXPECT_EQ(dif_kmac_init(dev().region(), &kmac_), kDifOk); }
+
+  /**
+   * Set mmio write expectation for 8 bits data size.
+   *
+   * @param message Buffer with the data.
+   * @param size Len of the buffer.
+   */
+  void setExpectedMessageByte(const uint8_t *message, const size_t size = 8) {
+    for (int i = 0; i < size; ++i) {
+      EXPECT_WRITE8(KMAC_MSG_FIFO_REG_OFFSET, message[i]);
+    }
+  }
+
+  /**
+   * Set mmio write expectation for 32 bits data size considering an alignment
+   * of 32 bits.
+   *
+   * @param message Buffer with the data.
+   * @param size Len of the buffer.
+   */
+  void setExpectedMessageInt32(const uint8_t *message, const size_t size = 8) {
```

Comment:
I'd rather we didn't set a default for the size parameter here. It would be easy to accidentally not set the size and end up with the 8 default. Same above.